scrapy/downloadermiddlewares/cookies.py

```diff
@@ -43,10 +43,11 @@ def _process_cookies(self, cookies, *, jar, request):
 
             if cookie_domain and _is_public_domain(cookie_domain):
                 if cookie_domain != request_domain:
-                    continue
-                cookie.domain = request_domain
-
-            jar.set_cookie_if_ok(cookie, request)
+                    cookie.domain = request_domain
```

Comment:
@wRAR With this modification the cookie is always set on the jar. The cookie domain must be the same as the request one in order to be set by `jar.set_cookie_if_ok()`

Reply:
I'm trying to understand why the logic is inverted by this change.